samza-sql/src/main/java/org/apache/samza/sql/planner/QueryPlanner.java

```diff
@@ -141,20 +142,28 @@ public RelRoot plan(String query) {
           .sqlToRelConverterConfig(SqlToRelConverter.Config.DEFAULT)
           .traitDefs(traitDefs)
           .context(Contexts.EMPTY_CONTEXT)
-          .costFactory(null)
+          .programs(
+              Programs.hep(ImmutableList.of(FilterJoinRule.FILTER_ON_JOIN), true, DefaultRelMetadataProvider.INSTANCE))
```

Comment:
Could we remove FILTER_ON_JOIN optimization ? It doesn't work well with remote joins. We should instead use the optimization for remote joins in the other PR that I sent.

Reply:
@atoomula this rule is not really the main part of this work and I don't think we need to fix/choose now which rule should be used to handle filters or should or should not be pushed to table scan this is beyond the scope of this PR. 
The main goal is to explore how we can handle the **operator stack between the join and remote table scan**. 
I added the rule to ensure that if a filter is there things works as expected and it seems it is working. But I would love to know what are the **blind spot(s)** I am missing about what work and does not work well with the remote table scan if you think this will help drive this issue. Thanks